### PR TITLE
Handling null UnloadAsset hook

### PR DESCRIPTION
**Pull request by KubeRoot**

### CODE REVIEW

File: UnityEngine.Base.mm/src/patch_Resources.cs [hunk 1]
```diff
@@ -42,7 +42,8 @@ internal sealed class patch_Resources {
         [MonoModOriginal] public static extern void OnloadAsset(Object assetToUnload);
         [MonoModOriginalName("OnloadAsset")]
         public static void UnloadAsset(Object asset) {
-            if (ETGModUnityEngineHooks.UnloadAsset(asset)) {
+            bool? overrideUnload = ETGModUnityEngineHooks.UnloadAsset?.Invoke(asset);
+            if (overrideUnload.HasValue && overrideUnload.Value) {
```
Comment (0x0ade) [inline UnityEngine.Base.mm/src/patch_Resources.cs]: Could be simplified as
```cs
if (ETGModUnityEngineHooks.UnloadAsset?.Invoke(asset) ?? false)
```

Where it either gives you false (don't skip) by default, or whatever the hook returns. Your method is working perfectly fine, though.

**Comment by 0x0ade** [comment]
Oh. Already merged...

**Comment by zatherz** [comment]
@0x0ade was only a test PR. Feel free to revert.

**Comment by 0x0ade** [comment]
@zatherz it's fine, hence why I gave it a "LGTM" (looks good to me).

I don't know if you knew, but it's generally good practice to wait for approval from the one responsible for the pieces of code the PR affects, or the project maintainers in general. And GitHub offers a way to review pull review.

I just didn't know it was meant as a "test PR" because it actually fixes possible issues with the hook being null. Once I get back to ETGMod (maybe next week, maybe never), I'll add similar measurements to all UnityEngine hooks.

**Pull request merged**